src/core/detection.cpp

```diff
@@ -0,0 +1,12 @@
+#include "detection.h"
+
+Detection::Detection(const std::string &type_,
```

Comment:
Se hai solo questo, e con questo modo di inizializzazione, puoi farlo direttamente nel `.h`. Non si deve eccedere, ma in casi come questo ha senso.

Reply:
ok, fatto!